neuralprophet/utils.py

```diff
@@ -987,3 +987,153 @@ def configure_trainer(
     # config["replace_sampler_ddp"] = False
 
     return pl.Trainer(**config), checkpoint_callback
+
+
+def unpack_sliced_tensor(
+    sliced_tensor,
+    n_lags,
+    n_forecasts,
+    max_lags,
+    feature_indices,
+    config_lagged_regressors,
+    config_seasonality,
+):
+    sliced_tensor = sliced_tensor.detach().clone()
+    inputs = OrderedDict()
+    if max_lags > 0:
+        # Unpack time feature (time doesn't need further slicing)
+        start_idx, end_idx = feature_indices["time"]
+        time_offset = max_lags - n_lags
+        inputs["time"] = sliced_tensor[:, time_offset : time_offset + n_lags + n_forecasts, start_idx]
+
+        # Unpack lags feature
+        if "lags" in feature_indices:
+            lags_start_idx, lags_end_idx = feature_indices["lags"]
+            lags_offset = max_lags - n_lags
+            inputs["lags"] = sliced_tensor[:, lags_offset : lags_offset + n_lags, lags_start_idx]
+
+        # Unpack targets
+        if "targets" in feature_indices:
+            targets_start_idx, targets_end_idx = feature_indices["targets"]
+            inputs["targets"] = sliced_tensor[:, max_lags : max_lags + n_forecasts, targets_start_idx].unsqueeze(2)
```

Comment:
Can you add a comment, why the unsqueeze is needed here? (but not in lags or time)